src/rgw/radosgw-admin/radosgw-admin.cc

```diff
@@ -9100,6 +9123,78 @@ int main(int argc, const char **argv)
     }
   }
 
+  if (opt_cmd == OPT::DEDUP_STATS) {
+    using namespace rgw::dedup;
+    rgw::sal::RadosStore *store = dynamic_cast<rgw::sal::RadosStore*>(driver);
+    if (!store) {
+      cerr << "ERROR: this command can only work when the cluster has a RADOS "
+	   << "backing store." << std::endl;
+      return EPERM;
+    }
+
+    int ret = cluster::collect_all_shard_stats(store, formatter.get(), dpp());
+    if (ret == 0) {
+      formatter->flush(cout);
+    }
+    else {
+      cerr << "ERROR: Failed reading stat counters" << std::endl;
+    }
+    return ret;
+  }
+
+  if (opt_cmd == OPT::DEDUP_ABORT || opt_cmd == OPT::DEDUP_PAUSE || opt_cmd == OPT::DEDUP_RESUME) {
+    using namespace rgw::dedup;
+    urgent_msg_t urgent_msg;
+    if (opt_cmd == OPT::DEDUP_ABORT) {
+      urgent_msg = URGENT_MSG_ABORT;
+    }
+    else if (opt_cmd == OPT::DEDUP_PAUSE) {
+      urgent_msg = URGENT_MSG_PASUE;
+    }
+    else {
+      urgent_msg = URGENT_MSG_RESUME;
+    }
+
+    rgw::sal::RadosStore *store = dynamic_cast<rgw::sal::RadosStore*>(driver);
+    if (!store) {
+      cerr << "ERROR: this command can only work when the cluster has a RADOS "
+	   << "backing store." << std::endl;
```

Comment:
ditto

Reply:
This exact error message is used all over the file for this scenario (I counted 10 places using this error message)

Reply:
That's why I was asking if the checks could be ||d into one.